### PR TITLE
Add spdx bom import functionality

### PR DESCRIPTION
**Pull request by maxhbr**
This PR adds the BOM import functionallity for projects and releases.

# File used for testing:
[example.spdx.rdf](https://github.com/eclipse/sw360/files/3646100/example.spdx.rdf.txt)

## The corresponding tag/value looks like:
```
SPDXVersion: SPDX-2.1
DataLicense: CC0-1.0
DocumentNamespace: http://www.spdx.org/spdxdocs/8f141b09-1138-4fc5-aecb-fc10d9ac1eed
DocumentName: SBOM toy exaple
SPDXID: SPDXRef-DOCUMENT
Creator: Organization: NTIA Standards and Formats Workgroup
Created: 2019-08-31T11:29:46Z
Relationship: SPDXRef-DOCUMENT DESCRIBES SPDXRef-asoftware-v1.1

PackageName: asoftware
SPDXID: SPDXRef-asoftware-v1.1
PackageVersion: 1.1
PackageSupplier: Organization: Acme
PackageDownloadLocation: NOASSERTION
FilesAnalyzed: false
PackageChecksum: SHA1: 75068c26abbed3ad3980685bae21d7202d288317
PackageLicenseConcluded: NOASSERTION
PackageLicenseDeclared: NOASSERTION
PackageCopyrightText: NOASSERTION
Relationship: SPDXRef-asoftware-v1.1 CONTAINS SPDXRef-Browser-v2.1
Relationship: SPDXRef-asoftware-v1.1 CONTAINS SPDXRef-Buffer-v2.2

PackageName: Browser
SPDXID: SPDXRef-Browser-v2.1
PackageVersion: 2.1
PackageSupplier: Person: Bob
PackageDownloadLocation: NOASSERTION
FilesAnalyzed: false
PackageChecksum: SHA1: 94568c26abbed3ad3980685deaf1d7202d268314
PackageLicenseConcluded: NOASSERTION
PackageLicenseDeclared: NOASSERTION
PackageCopyrightText: NOASSERTION
Relationship: SPDXRef-Browser-v2.1 CONTAINS SPDXRef-CompressionEng-v3.1

PackageName: Buffer
SPDXID: SPDXRef-Buffer-v2.2
PackageVersion: 2.2
PackageSupplier: Organization: Bingo
PackageDownloadLocation: NOASSERTION
FilesAnalyzed: false
PackageChecksum: SHA1: 84568c26aabad3ad3980685beef1d7202d26831d
PackageLicenseConcluded: NOASSERTION
PackageLicenseDeclared: NOASSERTION
PackageCopyrightText: NOASSERTION

PackageName: CompressionEng
SPDXID: SPDXRef-CompressionEng-v3.1
PackageVersion: 3.1
PackageSupplier: Person: Carol
PackageDownloadLocation: NOASSERTION
FilesAnalyzed: false
PackageChecksum: SHA1: 63568c26aebad3ad398bb85ce1f1d7202d27731a
PackageLicenseConcluded: NOASSERTION
PackageLicenseDeclared: NOASSERTION
PackageCopyrightText: NOASSERTION
```

### CODE REVIEW

**Comment by kallesoranko** [comment]
@maxhbr What is the status of this PR? Is it tested in PR testing or locally? If I remember correctly, the use case for this has been to integrate with vmware/tern using SPDX files, but does it also mean that general SPDX file generated by any other tool should also work here?

**Comment by maxhbr** [comment]
Yes, this PR can be used to import any valid SPDX document containing BOM information.

**Comment by maxhbr** [comment]
This is currently broken and WIP

**Comment by maxhbr** [comment]
@mcjaeger This PR is now rebased

**Comment by mcjaeger** [comment]
it has EPL-1.0 headers, but that is OK.

File: backend/src-common/pom.xml [hunk 1]
```diff
@@ -36,4 +36,23 @@
             </plugin>
         </plugins>
     </build>
+    <dependencies>
```
Comment (mcjaeger) [inline backend/src-common/pom.xml]: there seem to be another (and maybe more) backend project(s) which is importing the same - cans this then be outdated?

Comment (maxhbr) [reply backend/src-common/pom.xml]: Yes, but that would make it an implicit dependency and `mvn dependency:analyze` would add a warning for that.

File: backend/src-common/src/main/java/org/eclipse/sw360/spdx/SpdxBOMImporter.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            return Optional.empty();
+        }
+    }
+
+    private Map<String, ReleaseRelationship> makeReleaseIdToRelationship(List<SpdxBOMImporterSink.Response> releases) {
+        return releases.stream()
+                .collect(Collectors.toMap(SpdxBOMImporterSink.Response::getId, SpdxBOMImporterSink.Response::getReleaseRelationship));
+    }
+
+    private Project creatProjectFromSpdxPackage(SpdxPackage spdxPackage) {
```
Comment (mcjaeger) [inline backend/src-common/src/main/java/org/eclipse/sw360/spdx/SpdxBOMImporter.java]: minor issue ...

File: backend/src/src-attachments/src/main/java/org/eclipse/sw360/attachments/AttachmentHandler.java [hunk 3]
```diff
@@ -52,6 +52,10 @@ public AttachmentContent makeAttachmentContent(AttachmentContent attachmentConte
 
     @Override
     public List<AttachmentContent> makeAttachmentContents(List<AttachmentContent> attachmentContents) throws TException {
+        for (AttachmentContent attachmentContent : attachmentContents) {
+            validateAttachment(attachmentContent);
+            assertIdUnset(attachmentContent.getId());
```
Comment (mcjaeger) [inline backend/src/src-attachments/src/main/java/org/eclipse/sw360/attachments/AttachmentHandler.java]: interesting, wondering if that would block other attachment cases

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/utils/includes/importBom.jspf [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                        statusDiv.show();
+                        r.upload();
+                        statusDiv.html("<h2>Uploading ...</h2>");
+                    });
+                    r.on('fileProgress', function (file) {
+                        console.log("fileProgress...");
+                    });
+                    r.on('fileSuccess', function (file) {
+                        console.log("fileSuccess...");
+                        statusDiv.html("<h2>Importing ...</h2>");
```
Comment (mcjaeger) [inline frontend/sw360-portlet/src/main/resources/META-INF/resources/html/utils/includes/importBom.jspf]: This "Importing..." and "Import Success" is actully too much left and should be at dame distance form the edge as the text line above ... 

Screenshot 2020-02-19 at 20 12 20

**Comment by mcjaeger** [comment]
CI build is ending successfully, but has exited uncleanly

**Pull request merged**